File: OP2_labs/lab4_py/Person.py
from datetime import date
from dataclasses import dataclass

@dataclass
class Birthday:
    year : int
    month : int
    day : int

class Person(object):
    def __init__(self, row):
        birthdayNums = row[row.rfind(' ') + 1 : ].split('.')
        self.birthday = Birthday(int(birthdayNums[2]), int(birthdayNums[1]), int(birthdayNums[0]))
        self.name = row[ : row.rfind(' ')]

    def getAge(self):
        now = str(date.today()).split('-')
        now = Birthday(int(now[0]), int(now[1]), int(now[2]))
        if now.day < self.birthday.day:
            now.month -= 1
        if now.month < self.birthday.month:
            now.year -= 1
        return now.year - self.birthday.year

    def getName(self):
        return self.name

File: OP2_labs/lab4_cpp/Person.cpp
#pragma once
#include "Person.h"

int Person::getAge()
{
	time_t secondsNow = time(0);
	tm allNow;
	localtime_s(&allNow, &secondsNow);
	Date now;
	now.year = 1900 + allNow.tm_year;
	now.month = 1 + allNow.tm_mon;
	now.day = allNow.tm_mday;
	if (now.day < birthday.day) { now.month--; }
	if (now.month < birthday.month) { now.year--; }
	return now.year - birthday.year;
}
string Person::getName()
{
	return name;
}